mlflow/utils/autologging_utils/safety.py

```diff
@@ -447,6 +453,18 @@ def safe_patch_function(*args, **kwargs):
                     and autologging_integration
                     not in mlflow.utils.autologging_utils._AUTOLOGGING_GLOBALLY_DISABLED_EXEMPTIONS
                 )
+                or (
+                    _AUTOLOGGING_THREAD_LOCALLY_DISABLED.get()
+                    and autologging_integration
+                    not in _AUTOLOGGING_THREAD_LOCALLY_DISABLED_EXEMPTIONS.get()
+                )
+                or (
+                    # For typical use-case of "hyper parameter tuning", Optuna might fork multiple
```

Comment:
I feel like we should have some very simple test that can validate this where we run an optuna sweep for just a few iterations trying to optimize a simple sklearn model just to assert that we're getting the expected number of runs generated from autologging. Can we follow up later with this just to be safe?

Reply:
The unit tests can guard this case well.
but I think we can add a complete e2e test for optuna tuning, and check all runs / logged data / parent run are correctly generated.

Reply:
I think it's worth adding a realistic use case.

Reply:
For the multi-threading use case, can we create a docs page that covers this topic with specific recommendation (don't do x, do y, and here's why)? 
For multiprocessing, the inheritance model seems intuitive and it is the behavior that I would expect if I were a user who didn't read any docs on the topic. We should also mention this behavior in the docs and provide an instruction on how to change the behavior if desired.

Reply:
sure. I will file a follow-up PR for the doc page.